Prevent early exit on killall error in dev-env script

vitorenesduarte opened the pull request:
<!-- < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < < ☺
v                               ✰  Thanks for creating a PR! ✰    
v    Before smashing the submit button please review the checkboxes.
v    If a checkbox is n/a - please still include it but + a little note why
☺ > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > > >  -->

## Description

The `dev-env` script was exiting early if one of the `killall` commands returned an error.
This PR ignores any errors on those commands with `|| true` but I'm not sure this is the best solution.
Since there's a `set -e` right after the `killall` commands, maybe a better solution would be to remove the `-e` from `#!/bin/bash -e`.

<!-- Add a description of the changes that this PR introduces and the files that
are the most critical to review.
-->


______

For contributor use:

- [ ] Updated the __Unreleased__ section of [CHANGELOG.md](https://github.com/informalsystems/ibc-rs/blob/master/CHANGELOG.md) with the issue.
- [ ] If applicable: Unit tests written, added test to CI.
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Updated relevant documentation (`docs/`) and code comments.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

ancazamfir commented on dev-env:
```diff
@@ -50,8 +50,8 @@ echo "GAIA VERSION INFO: $(gaiad version)"
+killall gaiad &> /dev/null || true
+killall akash &> /dev/null || true
 
 set -e
```
I think we can remove this:
```suggestion
```

ancazamfir commented:
> ...maybe a better solution would be to remove the -e from #!/bin/bash -e.
 
I think we should leave that there. In general we want to exit if there are errors, except for some commands (the ones marked `|| true`), e.g if there were no previous gaia instances (addressed in this PR), no previous light client configuration.

ancazamfir commented:
Also I just noticed we don't check the number of parameters. Maybe add something like:
```
if [ "$#" -ne 3 ]; then
    echo "Incorrect number of parameters"
    echo "Usage: $0 [CONFIG_FILE] [CHAIN_ID_0] [CHAIN_ID_1]"
    exit 1
fi
```

vitorenesduarte commented:
@ancazamfir I've addressed the comments and I added a verification similar to what was suggested. The script now prevents the user to supply more arguments than needed.

ancazamfir commented:
> @ancazamfir I've addressed the comments and I added a verification similar to what was suggested. The script now prevents the user to supply more arguments than needed.

Thanks! One more comment since you're doing a nice cleanup on this, could you also add a check that the file exists.

adizere commented on dev-env:
```diff
@@ -3,28 +3,42 @@
 chainid1="$3"
 
+if ! [ -f $tomlcfg ]; then
+  echo "[CONFIG_FILE] does not exist or is not a file."
```
```suggestion
  echo "[CONFIG_FILE] ($1) does not exist or is not a file."
```

ancazamfir reviewed (approved):
Looks great!
